File: sdks/javascript/examples/basic_usage.js
import { NexusClient } from '../src/index.js';

const client = new NexusClient({ baseUrl: 'http://localhost:5000', tenantId: 'tenant-default' });

try {
  const health = await client.health();
  console.log('Health:', JSON.stringify(health, null, 2));

  const agents = await client.listAgents(1, 5);
  console.log(`Total agents: ${agents.totalCount}`);
  for (const a of agents.items.slice(0, 3)) {
    console.log(`  - ${a.id}: ${a.name} (${a.capability})`);
  }

  const swarms = await client.listSwarms();
  console.log(`Total swarms: ${swarms.totalCount}`);

  const kyber = await client.kyberKeypair();
  const pubKeyBytes = Buffer.from(kyber.publicKey, 'base64');
  console.log(`Kyber public key: ${pubKeyBytes.length} bytes`);
} catch (err) {
  console.error('Error:', err.message);
}

File: sdks/python/examples/basic_usage.py
"""Basic usage example for the Nexus Python SDK."""
import base64
from nexus_sdk import NexusClient

client = NexusClient(base_url="http://localhost:5000", tenant_id="tenant-default")

# Health check
health = client.health()
print("Health:", health)

# List agents
agents = client.list_agents(page=1, page_size=10)
print(f"Total agents: {agents.get('totalCount', 0)}")
for agent in agents.get("items", [])[:3]:
    print(f"  - {agent['id']}: {agent['name']} ({agent['capability']})")

# List swarms
swarms = client.list_swarms()
print(f"Total swarms: {swarms.get('totalCount', 0)}")

# Kyber key generation
kyber = client.kyber_keypair("Kyber768")
print(f"Kyber public key size: {len(base64.b64decode(kyber['publicKey']))} bytes")

# Run Shamir split
import base64
secret_b64 = base64.b64encode(b"SuperSecret42!").decode()
shares = client.shamir_split(secret_b64, threshold=3, total=5)
print(f"Shamir shares created: {shares.get('shareCount', 0)}")
